pandas/tools/merge.py

```diff
@@ -894,13 +896,21 @@ def get_result(self):
             if self.axis == 0:
                 new_data = com._concat_compat([x.values for x in self.objs])
                 name = com._consensus_name_attr(self.objs)
-                return Series(new_data, index=self.new_axes[0], name=name).__finalize__(self, method='concat')
+                if self._is_sp_series:
```

Comment:
I was a little fast on the trigger. I didn't realize that you need `._is_frame` etc elsewhere.

For this code, this is what I was originally thinking. Tell me if I'm still off, though

In `__init__`
```python
self.obj_constructor=sample._constructor
```
and then in `get_result`
```python
self.obj_constructor(...
```